Page  otimization api

Jhonattan2121 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Improved tab loading and caching for a smoother and faster experience when navigating spaces and tabs.
  - Added a method to clear locally cached spaces without affecting other user data.

- **Bug Fixes**
  - Loading skeletons now display more accurately, only appearing during initial space loads or when switching spaces.

- **Refactor**
  - Reorganized analytics event definitions for better consistency and type safety.
  - Updated axios API instance configuration for improved flexibility and reliability.
  - Improved import organization and removed unused code for better maintainability.

- **Style**
  - Enhanced semantic HTML by updating navigation elements for accessibility.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Jhonattan2121 committed:
fix: Removed dynamic key from SpacePage to avoid infinite requests and unnecessary remounts

Jhonattan2121 committed:
fix: fix commit

Jhonattan2121 committed:
perf: optimize unnecessary requests in fidgets

Jhonattan2121 committed:
fix: fix React hooks bug by moving useCallback and useMemo to the main component scope

Jhonattan2121 committed:
fix: correction build notification

Jhonattan2121 committed:
fix: Fixed infinite re-rendering issue

Jhonattan2121 committed:
Refactor: Centralize Axios instance for API calls optimization

willyogo commented on src/app/(spaces)/PublicSpace.tsx:
```diff
@@ -164,60 +164,56 @@ export default function PublicSpace({
-    localSpaces,
+    setCurrentSpaceId,
+    setCurrentTabName,
   ]);
```
possibly fixed in this PR? https://github.com/Nounspace/nounspace.ts/pull/1139

Jhonattan2121 committed:
Fix image flashing on partial visibility & improve feed/app performance by minimizing resource load

Jhonattan2121 committed:
Merge branch 'canary' into page--otimization-api

Jhonattan2121 committed:
feat: adds cache to avoid repeated calls in loadUserSpaceData

- Implemented a cache mechanism in the  function to avoid repeated calls with the same parameters.
- Reduces unnecessary requests and improves performance.
- Includes logs to identify cache hits.

Jhonattan2121 committed:
fix: Fixes local cache issue in PublicSpace component and adjusts AppStore typing

Jhonattan2121 committed:
fix: ensures unsaved changes are discarded when reloading public space pages

Fix an issue where unsaved (locally cached) changes were persisted when reloading the space page. Main changes:

- Implements local cache clearing when loading a public space
- Fixes rendering issues that caused flickering when adding/removing fidgets
- Optimizes memoization to avoid unnecessary re-renders
- Ensures that data is always loaded from the database after page reloads

Jhonattan2121 committed:
fix: fix tabs loading

Jhonattan2121 committed:
fix: preserve account/homebase data when clearing spaces

Jhonattan2121 committed:
fix: add fallback types for analytics events

Jhonattan2121 committed:
fix: add root to useIntersectionObserver dependencies
